File: Problems_and_Solutions/days-10 to 19/day-11/Java/Solution1.java
import java.util.Arrays;
class Solution1 {
    public static void bubbleSort(int[] arr, int n) {
        if (n == 1) return;

        for(int i=0; i<n; i++){
            if(arr[i] > arr[i+1]){
                int temp = arr[i];
                arr[i] = arr[i+1];
                arr[i+1] = temp;
            }
        }
        
        bubbleSort(arr,n-1);
    }

    public static void main(String[] args){
        int a[] = {4, 1, 3, 9, 7};
        bubbleSort(a,a.length-1);
        System.out.println(Arrays.toString(a));
    }
}

File: Problems_and_Solutions/days-10 to 19/day-11/cpp/solution 1.cpp
class Solution {
  public:
    // Function to sort the array using bubble sort algorithm.
    void bubbleSort(vector<int>& arr) {
        rec1(arr,0);
        
    }
    void rec1(vector<int>& arr,int i){
        if(i<arr.size())
        {
            rec2(arr,i,0);
            rec1(arr,i+1);
        }
        
    }
    void rec2(vector<int>& arr,int i,int j){
        if(j<arr.size()-1-i)
        {
            if(arr[j]>arr[j+1])
            {
                int temp=arr[j];
                arr[j]=arr[j+1];
                arr[j+1]=temp;
            }
            rec2(arr,i,j+1);
        }
    }
};